native/cocos/platform/openharmony/modules/SystemWindow.cpp

```diff
@@ -0,0 +1,152 @@
+
+void OnSurfaceCreatedCB(OH_NativeXComponent* component, void* window) {
+    SendToWorker(cc::MessageType::WM_XCOMPONENT_SURFACE_CREATED, component, window);
+}
+
+void DispatchTouchEventCB(OH_NativeXComponent* component, void* window) {
+    SendToWorker(cc::MessageType::WM_XCOMPONENT_TOUCH_EVENT, component, window);
+}
+
+void OnSurfaceChangedCB(OH_NativeXComponent* component, void* window) {
+    SendToWorker(cc::MessageType::WM_XCOMPONENT_SURFACE_CHANGED, component, window);
+}
+
+void OnSurfaceDestroyedCB(OH_NativeXComponent* component, void* window) {
+    SendToWorker(cc::MessageType::WM_XCOMPONENT_SURFACE_DESTROY, component, window);
+}
+} // namespace
+
+namespace cc {
+SystemWindow* SystemWindow::instance_ = nullptr;
+
+SystemWindow* SystemWindow::GetInstance() {
+    return instance_;
+}
+
+SystemWindow::SystemWindow() {
+    instance_                    = this;
+    callback_.OnSurfaceCreated   = OnSurfaceCreatedCB;
+    callback_.OnSurfaceChanged   = OnSurfaceChangedCB;
+    callback_.OnSurfaceDestroyed = OnSurfaceDestroyedCB;
+    callback_.DispatchTouchEvent = DispatchTouchEventCB;
+}
+
+bool SystemWindow::createWindow(const char* title,
+                                int x, int y, int w,
+                                int h, int flags) {
+    CC_UNUSED_PARAM(title);
+    CC_UNUSED_PARAM(x);
+    CC_UNUSED_PARAM(y);
+    CC_UNUSED_PARAM(w);
+    CC_UNUSED_PARAM(h);
+    CC_UNUSED_PARAM(flags);
+    return true;
+}
+
+void SystemWindow::SetNativeXComponent(OH_NativeXComponent* component) {
+    component_ = component;
+    OH_NativeXComponent_RegisterCallback(component_, &callback_);
+}
+
+void SystemWindow::setCursorEnabled(bool value) {
+}
+
+void SystemWindow::copyTextToClipboard(const std::string& text) {
+    //copyTextToClipboardJNI(text);
+}
+
+uintptr_t SystemWindow::getWindowHandler() const {
+    return reinterpret_cast<uintptr_t>(windowHandler_);
+}
+
+SystemWindow::Size SystemWindow::getViewSize() const {
+    return Size{static_cast<float>(width_),
+                static_cast<float>(height_)};
+}
+
+void SystemWindow::OnSurfaceCreated(OH_NativeXComponent* component, void* window) {
+    int32_t ret = OH_NativeXComponent_GetXComponentSize(component, window, &width_, &height_);
+    CCASSERT(ret == OH_NATIVEXCOMPONENT_RESULT_SUCCESS, "GetXComponentSize call failed");
+    windowHandler_ = window;
+}
+
+void SystemWindow::OnSurfaceChanged(OH_NativeXComponent* component, void* window) {
+}
+
+void SystemWindow::OnSurfaceDestroyed(OH_NativeXComponent* component, void* window) {
+    windowHandler_ = nullptr;
+}
+
+void SystemWindow::DispatchTouchEvent(OH_NativeXComponent* component, void* window) {
+    struct OH_NativeXComponent_TouchEvent touchEvent;
+    int32_t ret = OH_NativeXComponent_GetTouchEvent(component, window, &touchEvent);
+    if (ret != OH_NATIVEXCOMPONENT_RESULT_SUCCESS) {
+        return;
+    }
+
+    OpenHarmonyPlatform* platform = dynamic_cast<OpenHarmonyPlatform*>(BasePlatform::getPlatform());
```

Comment:
static_cast is enough.

Reply:
Base class to derived class, it should be better to use dynamic_cast. Semantics are clear